library/oneview_firmware_driver.py

```diff
@@ -141,11 +121,35 @@ def execute_module(self):
 
     def __present(self, resource):
         if not resource:
-            resource = self.oneview_client.firmware_drivers.create(self.data)
+            data = self.__parse_data()
+            resource = self.oneview_client.firmware_drivers.create(data)
             return True, self.MSG_CREATED, dict(firmware_driver=resource)
         else:
             return False, self.MSG_ALREADY_PRESENT, dict(firmware_driver=resource)
 
+    def __parse_data(self):
+        data = self.data.copy()
+        # Allow usage of baselineName instead of baselineUri
+        if data and data.get('baselineName'):
+            baseline_name = data.pop('baselineName', "")
+            spp = self.get_by_name(baseline_name)
+            if spp:
+                data['baselineUri'] = spp['uri']
+            else:
+                raise HPOneViewException('Baseline SPP named "%s" not found in OneView Appliance.' % baseline_name)
+
+        # Allow usage of hotfixNames instead of hotfixUris
+        if data and data.get('hotfixNames'):
+            hotfix_names = data.pop('hotfixNames', [])
+            data['hotfixUris'] = data.get('hotfixUris') or []
```

Comment:
No need to change, but you could have used the same as above: `.get('hotfixUris', [])`